packages/algob/src/lib/tx.ts

```diff
@@ -247,3 +247,119 @@ export async function transferASALsig (
+  if (transactionResponse['global-state-delta'] !== undefined) {
+    console.log("Global State updated:", transactionResponse['global-state-delta']);
+  }
+  if (transactionResponse['local-state-delta'] !== undefined) {
+    console.log("Local State updated:", transactionResponse['local-state-delta']);
+  }
+}
+
+/**
+ * Description: The user may discontinue use of the application by sending a close out transaction.
+ * This will remove the local state for this application from the user's account.
+ * @param deployer AlgobDeployer
+ * @param sender Account from which call needs to be made
+ * @param payFlags Transaction Flags
+ * @param appId ID of the application being configured or empty if creating
+ */
+export async function closeOutSSC (
+  deployer: AlgobDeployer,
+  sender: AccountSDK,
+  payFlags: TxParams,
+  appId: number): Promise<void> {
+  const params = await mkSuggestedParams(deployer.algodClient, payFlags);
+
+  const txn = tx.makeApplicationCloseOutTxn(sender, params, appId);
+
+  const txId = txn.txID().toString();
+  const signedTxn = txn.signTxn(sender.sk);
+  await deployer.algodClient.sendRawTransaction(signedTxn).do();
+  await deployer.waitForConfirmation(txId);
+}
+
+/**
+ * Description: The approval program defines the creator as the only account able to delete the application.
+ * This removes the global state, but does not impact any user's local state.
+ * @param deployer AlgobDeployer
+ * @param sender Account from which call needs to be made
+ * @param payFlags Transaction Flags
+ * @param appId ID of the application being configured or empty if creating
+ */
+export async function deleteTxnSSC (
+  deployer: AlgobDeployer,
+  sender: AccountSDK,
+  payFlags: TxParams,
+  appId: number): Promise<void> {
+  const params = await mkSuggestedParams(deployer.algodClient, payFlags);
+
+  const txn = tx.makeApplicationDeleteTxn(sender, params, appId);
+
+  const txId = txn.txID().toString();
+  const signedTxn = txn.signTxn(sender.sk);
+  await deployer.algodClient.sendRawTransaction(signedTxn).do();
+  await deployer.waitForConfirmation(txId);
+}
+
+/**
+ * Description: The user may clear the local state for an application at any time,
+ * even if the application was deleted by the creator.
+ * @param deployer AlgobDeployer
+ * @param sender Account from which call needs to be made
+ * @param payFlags Transaction Flags
+ * @param appId ID of the application being configured or empty if creating
+ */
+export async function clearStateSSC (
+  deployer: AlgobDeployer,
+  sender: AccountSDK,
+  payFlags: TxParams,
+  appId: number): Promise<void> {
+  const params = await mkSuggestedParams(deployer.algodClient, payFlags);
+
+  const txn = tx.makeApplicationClearStateTxn(sender, params, appId);
+
+  const txId = txn.txID().toString();
+  const signedTxn = txn.signTxn(sender.sk);
+  await deployer.algodClient.sendRawTransaction(signedTxn).do();
+  await deployer.waitForConfirmation(txId);
+}
```

Comment:
shouldn't all this functions be deployer methods? What do you think?

Reply:
I thought about this, after reading this 
```
Calls to stateful smart contracts are implemented using ApplicationCall transactions. These transactions types are as follows:

    NoOp - Generic application calls to execute the ApprovalProgram
    OptIn - Accounts use this transaction to opt in to the smart contract to participate (local storage usage).
    DeleteApplication - Transaction to delete the application.
    UpdateApplication - Transaction to update TEAL Programs for a contract.
    CloseOut - Accounts use this transaction to close out their participation in the contract. This call can fail based on the TEAL logic, preventing the account from removing the contract from its balance record.
    ClearState - Similar to CloseOut, but the transaction will always clear a contract from the account’s balance record whether the program succeeds or fails.
```
these funtions are application calls that are interacting with contract, but deployer deploys the contracts.
I still have doubt about `UpdateApllication` function, where should i put this ? what do you suggest ?
- Make a `application.ts` with all these calls
- Put all functions in tx.ts
- Put all functions in  tx.ts except UpdateApplication (put this in deployer)
- or something else
what do you think ?

Reply:
I would create `ssc.ts` and put that functions there, and remove `SSC` infix` from the name.

Reply:
yes, seems better